rotkehlchen/premium/sync.py

```diff
@@ -197,6 +197,36 @@ def maybe_upload_data_to_server(self) -> None:
         self.data.db.update_last_data_upload_ts(self.last_data_upload_ts)
         log.debug('upload to server -- success')
 
+    def sync_data(self, action: Literal['upload', 'download']) -> Tuple[bool, str]:
+        assert self.premium, 'This function has to be called with a not None premium'
+
+        if action == 'upload':
+            b64_encoded_data, our_hash = self.data.compress_and_encrypt_db(self.password)
```

Comment:
For both upload and download I would do it a bit differently here to not duplicate code.

Edit `maybe_upload_data_to_server()` and add a `force_upload` boolean which is `False` by default but when pushed through here, it essentially ignores all checks.

This way we can reuse the same function.

Same for pulling data. We have `_sync_data_from_server_and_replace_local()`

Reply:
I thought of the same in the beginning but since I wanted to also pass the messages I have to change the signature a bit. Is that ok? one of the methods returns `None` and the other `bool` to reuse along with `force_upload` I will have to change the return to `Tuple[bool, str]` so I can push the messages to the api caller.

Reply:
What messages do you need to pass? Most of the messages are shown to the API user by processing the raised exception and reading its error on the topmost part of the API server.

Looking at the function this is the only place you return a message:

```python
return False, 'No data found.'
```

Why not then just assume False means no data found. As all other paths seem to either raise, or return True

Reply:
Sure I can do that :)

Reply:
Ok another question, in both the functions, since they were not designed for external (rest api) use, the `RemoteError` is handled internally and it is not raised outside the method. 

Would it make sense to modify the method so that we handle the exception outside of the method in any case?